src/main/java/org/commcare/formplayer/application/FormController.java

```diff
@@ -198,27 +196,28 @@ public SubmitResponseBean submitForm(@RequestBody SubmitRequestBean submitReques
                         purgeCasesTimer.durationInMs() > 2 ?
                                 "Purging cases took some time" : "Probably didn't have to purge cases", extras);
 
-                ResponseEntity<String> submitResponse = submitService.submitForm(
-                        formEntrySession.getInstanceXml(),
-                        formEntrySession.getPostUrl()
-                );
-
-                if (!submitResponse.getStatusCode().is2xxSuccessful()) {
-                    if (submitResponse.getStatusCode() == HttpStatus.TOO_MANY_REQUESTS) {
-                        submitResponseBean.setStatus(Constants.SUBMIT_RESPONSE_TOO_MANY_REQUESTS);
-                    } else {
-                        submitResponseBean.setStatus("error");
-                        NotificationMessage notification = new NotificationMessage(
-                                "Form submission failed with error response" + submitResponse,
-                                true, NotificationMessage.Tag.submit);
-                        submitResponseBean.setNotification(notification);
-                        logNotification(notification, request);
-                        log.error("Submit response bean: " + submitResponseBean);
-                    }
+                String response;
+                try {
+                    response = submitService.submitForm(
+                            formEntrySession.getInstanceXml(),
+                            formEntrySession.getPostUrl()
+                    );
+                } catch (HttpClientErrorException.TooManyRequests e) {
+                    submitResponseBean.setStatus(Constants.SUBMIT_RESPONSE_TOO_MANY_REQUESTS);
+                    return submitResponseBean;
+                } catch (HttpClientErrorException e) {
+                    submitResponseBean.setStatus("error");
+                    NotificationMessage notification = new NotificationMessage(
+                            String.format("Form submission failed with error response: %s, %s, %s",
+                                    e.getMessage(), e.getResponseBodyAsString(), e.getResponseHeaders()),
+                            true, NotificationMessage.Tag.submit);
+                    submitResponseBean.setNotification(notification);
+                    logNotification(notification, request);
+                    log.error("Submit response bean: " + submitResponseBean);
                     return submitResponseBean;
                 }
```

Comment:
This changed to use the normal rest template instead of the one that passed errors through. The only real change is that we're handling exceptions instead of checking response codes.